src/extension/op_def.rs

```diff
@@ -96,10 +95,14 @@ pub(super) enum SignatureFunc {
     // CustomSignatureFunc trait too, and replace this enum with Box<dyn CustomSignatureFunc>.
     // However instead we treat all CustomFunc's as non-serializable.
     #[serde(rename = "signature")]
-    TypeScheme(OpDefTypeScheme),
+    TypeScheme(PolyFuncType),
     #[serde(skip)]
     CustomFunc {
-        params: Vec<TypeParam>,
+        /// All type parameters (both those passed to [func] and any passed
+        /// to the PolyFuncType returned by that)
+        all_params: Vec<TypeParam>,
```

Comment:
I considered having two separate vectors here, but we are constrained by having to implement `fn params(&self) -> &[TypeParam]` so we need this vec to return a view/slice thereof

Reply:
I find this specification unintuitive. If we went with "declare params for binary only" those would be the only ones here right?

Reply:
Yes. I've done this now. Everything is much neater - but the downside is, supposing you are writing a Hugr using an LSP in your editor or similar - there is now no method to say, "here are the static/binary params, what more params do I need?)". This is clearly something we will want at some stage - at the moment I'm leaving that until someone embarks upon such an LSP, but I could add something now if we like?